### ISSUE TITLE
"Done & Talk" button loading the next-to-classify subject in Talk instead of the just-classified subject

### ISSUE THREAD

**Issue by vrooje** [zooniverse/Panoptes-Front-End#3679]
## Expected behavior

When I click "Done & Talk" in the classification interface, the subject that loads in Talk is the subject I just classified.

## Current behavior

The subject that loads when I click "Done & Talk" seems to be the subject I'm *about to be served next* instead of the one I've just finished by clicking "Done & Talk". 

Not sure if this is always or only when I take these specific steps...

## Steps to replicate

1. Classify a subject (call it subject A) on [Planet 9 (SGL)](https://www.zooniverse.org/projects/skymap/planet-9/classify).
2. When finished classifying, click "Done & Talk" with ctrl/command (depending on OS) held down, so that Talk opens in a new tab (tab a), but the classification tab stays in focus.
3. Don't visit tab a right away; instead classify the new subject (subject B) and click "Done & Talk" to open in a new tab (tab b) again. Subject C will load in the classification interface, but don't classify it.
4. Click on tab a: it should be showing subject A, but for me it's showing subject B.
5. Click on tab b: it should be showing subject B, but for me it's showing subject C, which I haven't classified yet. 

## Additional information

- **Operating system:** Mac OS X El Capitan (10.11.6 (15G1108))
- **Browser:** Chrome Version 56.0.2924.87 (64-bit)

I'm also doing this from my home internet connection, which regularly loses connection for a minute or so at a time. (I need a new router.)

**Comment by srallen** [comment]
Noting that Planet 9 is using the hide classification summaries configuration, so when done or done & talk is clicked, it is saving the classification and then calling to load the next subject. A good guess as to what is happening is that the subject id in the button link is updating with the new subject id before the tab is opened and loads the URL.

**Comment by eatyourgreens** [comment]
Could be a browser bug, since the event handler on the link lets the browser handle the default event (opening the link URL.) Interesting clickjacking technique if the javascript is changing the URL before the browser loads the linked page.

It could also be a bug in `react-router`, since internal links should be intercepted by the router and pushed to location history, rather than reloading the page.

Linked to #3680 since fixing that issue will introduce this bug for the Talk link in the classification summary.

**Comment by yshish** [comment]
It happens to me too on AmazonCam project but only when using chromium browser. It doesn't happen when using Firefox.

**Comment by mschwamb** [comment]
A volunteer on P4: Ridges mentioned this as well, but I don't get the issue on my setup with Mac OS + Firefox. I'm asking for their setup.

**Comment by mschwamb** [comment]
Info from the volunteer:

I am using Safari on a Mac running OS 10.11.6. Also happens with Chrome.

**Comment by eatyourgreens** [comment]
Just to clarify: this bug only occurs when you ctrl-click to open the Talk page in a new tab?

**Comment by yshish** [comment]
@eatyourgreens It happens always when I click on the "Done & Talk" with the middle mouse button. When I click on it with the right mouse button and choose to open the link in a new tab from the list, it works without problems.

**Comment by eatyourgreens** [comment]
Confirmed here on OSX: 
**cmd-click**: opens a new tab with the next subject displayed
**ctrl-click**: 'Open in a new tab' opens a new tab with the current subject displayed.

**Comment by yshish** [comment]
(It happens to me while using Ubuntu 64-bit)

**Comment by srallen** [comment]
@yshish is this still an issue?

**Comment by yshish** [comment]
Nope, not any more.

**Comment by yshish** [comment]
I just wonder what happens when someone clicks on the 'Done and talk" button by the way to open it in a new browser tab, and then clicks again on the 'Done' button to get a new subject to classify, does it record the second click as another (second) classifications of that subject, or does it count just the first click?

**Comment by srallen** [comment]
Thanks @yshish. I could dig through the code, but I think we fixed this bug, but unfortunately this issue didn't get tagged when the fix was made. 

What should happen regardless of which button is hit is that the classification is submitted and the next subject is loaded. Done & Talk will navigate to the subject's Talk page that you just submitted the classification for. If this is behaving as it should, even if you click Done & Talk to open a new browser tab, if you go back to the tab with the classify page, you should see the next subject and thus it should be impossible to submit the same classification again.

**Comment by yshish** [comment]
@srallen What happens to me when I click on the Done&Talk by the middle mouse button - the image stays on the page as it was, offering me to click on the two buttons (Done and Talk and Done), but in a new browser tab it opens the image on the Talk. So I am technically allowed to click on the Done, or on the Done&Talk button again, and as many times as I want to. It doesn't load me a new subject to classify unless I click on either of these buttons again, using the left mouse button. (Or unless I open the classification in a new browser tab.) Even when I decide not to click on the button again and go on the project home page and back, it still shows me the last image and offers me to click on the Done buttons.

Hope it makes sense. I think this is not perfect. And I would like to know whether the repeated clicking on Done buttons does count as repeated classification or doesn't. Because it doesn't load a new image when I use the middle button to open the Talk in a new browser tab.

**Comment by yshish** [comment]
Anyway, this is much better than loosing the image I wanted to comment on, as was happening before the bug fix.

**Comment by srallen** [comment]
Ok, it sounds like this isn't quite solved yet. I'll reopen.

**Comment by yshish** [comment]
@camallen  

I have just realized that clicking on the Done&Talk button using the middle mouse button (to open the Talk in a new browser tab) could be reason why I received this subject https://www.zooniverse.org/projects/rainforestexpeditions/amazoncam-tambopata/talk/subjects/12646909 to classify for the second time today. 

I did use the middle mouse button to be able to comment on it on the Talk and to be allowed to use the zoom tools in the classification at the same time. I didn't succeed to ID the bird species but I remember all the details. After a couple of hours I got the same subject to classify again. 

I am 100% sure it is the same subject ID as I had it in my collection of birds already when I get it for the second time. And I do remember to be using the middle mouse button the first time.

So I wonder whether it recorded my first classification... Can someone look at it?
